Update Gutenberg JS packages [MAILPOET-4301]

lysyjan opened the pull request:
[MAILPOET-4301]

[MAILPOET-4301]: https://mailpoet.atlassian.net/browse/MAILPOET-4301?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

lysyjan committed:
Update @wordpress packages

[MAILPOET-4301]

lysyjan committed:
Update @types/wordpress__* packages

[MAILPOET-4301]

lysyjan committed:
Fix compatibility to allow compilation

[MAILPOET-4301]

lysyjan committed:
Fix family-font select design in popover

[MAILPOET-4301]

lysyjan committed:
Remove ColorSettings component

[MAILPOET-4301]

lysyjan committed:
Use newer Gutenberg component for settings colors

[MAILPOET-4301]

lysyjan committed:
Remove old color component usage

[MAILPOET-4301]

lysyjan committed:
Fix sidebar's family-font select

[MAILPOET-4301]

lysyjan committed:
Unify color panel design

[MAILPOET-4301]

costasovo reviewed (changes requested):
Excellent job! I really like the new inputs for colors! 👍 

I went through all blocks, and I found one issue we need to address.

The core/column block, which represents a column inside core/columns block, introduced a new setting called "Layout". It allows setting the width of content and width of the column.

We don't support this new settings in our data mapper [for the column block](https://github.com/mailpoet/mailpoet/blob/45e1db5893f46ce26a7a440fa74170380565fac6/mailpoet/assets/js/src/form_editor/store/blocks_to_form_body.jsx#L192-L216) (and also in the [mapper from form data to blocks](https://github.com/mailpoet/mailpoet/blob/trunk/mailpoet/assets/js/src/form_editor/store/form_body_to_blocks.jsx#L85-L161)) so it is not saved and also we don't support it in [the form column rederer on the frontend](https://github.com/mailpoet/mailpoet/blob/trunk/mailpoet/lib/Form/Block/Column.php).

![Screenshot 2022-08-03 at 10 36 45](https://user-images.githubusercontent.com/1082140/182565646-f96c247d-3325-4d55-a647-f962e8c4b787.png)

I think we have two options here:
1) Try to disable the new settings [when we register the column block](https://github.com/mailpoet/mailpoet/blob/trunk/mailpoet/assets/js/src/form_editor/blocks/columns/column.jsx#L16-L17). By modifying the metadata's support property.
2) Add support for the new setting.

I find the new feature quite handy, so I would suggest exploring the second option first.
It means adding code to a couple of places (mappers, mappers tests and front end renderer), but it should be quite straight forward.

lysyjan committed:
Disable layout configuration

[MAILPOET-4301]

lysyjan commented:
@costasovo Thank you for this catch!
I decided to disable this feature because I wasn't able to use the `Wide` configuration properly. Maybe we can recheck this feature during the next update.

samnajian committed:
Prioritize main over module for @woocommerce/components

Webpack was prioritizing module over main for
@woocommerce/components which was problematic

[MAILPOET-4507]

samnajian commented:
FYI: Just rebased trunk since had branched from this branch and the new branch

samnajian committed:
Fix issue with recursion ending up breaking watch:js command

[MAILPOET-4522]

samnajian committed:
Break compilation when compile:js hits a warning

[MAILPOET-4507]
